api/v1beta1/policyautomation_types.go

```diff
@@ -52,6 +54,25 @@ type AutomationDef struct {
 	TowerSecret string `json:"secret"`
 	// JobTTL sets the time to live for the Kubernetes AnsibleJob object after the Ansible job run has finished.
 	JobTTL *int `json:"jobTtl,omitempty"`
+	// The maximum number of violation contexts will be provided to the Ansible Tower as extra variable.
+	// +kubebuilder:validation:Minimum=0
+	PolicyViolationContextLimit uint `json:"policyViolationContextLimit,omitempty"`
+}
+
+// ViolationContext defines the non-compliance replicated policy information
+// that were sent to the Anisbile job through extra_var.
+type ViolationContext struct {
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
+	TargetClusters []string `json:"targetClusters,omitempty"`
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
+	PolicyName             string                                     `json:"policyName"`
+	Namespace              string                                     `json:"namespace,omitempty"`
```

Comment:
I think I'd rather not have `omitempty` on any of these. That way the Ansible user does not need to check if the key exists before checking the value.

Reply:
It'd be even better if it was a null value rather than an empty string:
https://stackoverflow.com/a/31048860